lucene/core/src/java/org/apache/lucene/index/StandardDirectoryReader.java

```diff
@@ -143,12 +158,17 @@ static StandardDirectoryReader open(
 
       writer.incRefDeleter(segmentInfos);
 
+      if (writer.getConfig().getLeafSorter() != null) {
+        readers.sort(writer.getConfig().getLeafSorter());
```

Comment:
Is IWC.leafSorter allowed to be live-updated?  If so, maybe we should pull out local variable here to hold onto the `leafSorter` and use that variable here and below, to ensure it is the same?

Reply:
@mikemccand  Thank you for your feedback.
The intent was to keep `leafSorter` constant and non-updatable, as  `indexWriter.getConfig` returns `LiveIndexWriterConfig` that doesn't allow setting of a new `leafSorter`. I understand that `leafSorter` in config can still be changed, and your comment is very valid. 

But In 8155f2e2abd4ed3a6280aae940d018dfe3b3dad1  I've moved sorting of leaves to the constructor, so hopefully this should resolve this concern.